source/server-discovery-and-monitoring/tests/unified/interruptInUse-pool-clear.yml

```diff
@@ -0,0 +1,322 @@
+              commandName: find
+          - commandStartedEvent:
+              commandName: find
+          - commandSucceededEvent:
+              commandName: find
+      - client: *client
+        eventType: cmap
+        events:
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {}
+          - connectionCheckedOutEvent: {}
+          - poolClearedEvent:
+              interruptInUseConnections: true
+          - connectionCheckedInEvent: {}
+          - connectionClosedEvent: {}
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {}
+     
+    outcome:
+      - collectionName: *collectionName
+        databaseName: *databaseName
+        documents:
+          - _id: 1
+  - description: Error returned from connection pool clear with interruptInUseConnections=true is retryable for write
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                useMultipleMongoses: false
+                observeEvents:
+                  - poolClearedEvent
+                  - connectionClosedEvent
+                  - commandStartedEvent
+                  - commandFailedEvent
+                  - commandSucceededEvent
+                  - connectionCheckedOutEvent
+                  - connectionCheckedInEvent
+                uriOptions:
+                  connectTimeoutMS: 500
+                  heartbeatFrequencyMS: 500
+                  appname: interruptInUseRetryableWrite
+                  retryWrites: true
+                  minPoolSize: 0
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+      # ensure the primary is discovered
+      - name: insertOne
+        object: *collection
+        arguments:
+          document: { _id: 1 }
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *setupClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode:
+              times: 3
+            data:
+              failCommands:
+                - hello
+                - isMaster
+                - insert
+              blockConnection:  true,
+              blockTimeMS: 2000,                
+              appName: interruptInUseRetryableWrite
+      # The failpoint will block the insert and next monitor check by 2 seconds each. 
+      # The monitor check should time out after 1 second plus whatever time was left 
+      # over on the prior heartbeat (max 500ms), cancelling the insert before it would succeed. 
+      # The retry will happen and succeed after the server is eventually rediscovered.
+      - name: insertOne
```

Comment:
Could we use an update with $where sleep() to match the other tests?

Reply:
good point, yes, I think it will be a good simplification, will push it tomorrow

Reply:
done